Assets/Mirror/Runtime/NetworkAuthenticator.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Mirror
+{
+    /// <summary>
+    /// Unity Event for the NetworkConnection
+    /// </summary>
+    [Serializable] public class UnityEventConnection : UnityEvent<NetworkConnection> { }
```

Comment:
maybe name it UnityEventNetworkConnection because it's a Unity event for the NetworkConnection? it's a bit longer, but more consistent

Reply:
Maybe AuthenticationEvent?

Reply:
if we keep it general like UnityEventNetworkConnection then we can reuse it later for other things. Otherwise we'd end up with multiple UnityEvent<NetworkConnection>.

that's also why I suggested naming it exactly what it is. I do this in my assets too. Just name it whatever UnityEvent<T> it is and then reuse where needed. keeps it simple.